tests/unit/versioning/VersionID.spec.js

```diff
@@ -9,83 +10,124 @@ function randkey(length) {
     return key;
 }
 
-describe('test generating versionIds', () => {
-    const count = 1000;
+function generateRandomVIDs(count) {
     const vids = Array(count).fill(null);
     for (let i = 0; i < count; i++) {
         vids[i] = VID.generateVersionId(randkey(15), 'PARIS');
     }
     process.env.VID_CRYPTO_PASSWORD = randkey(64);
+    return vids;
+}
+
+const count = 1000000;
+
+describe('test generating versionIds', () => {
+    describe('legaxy hex encoding', () => {
+        env.S3_VERSION_ID_ENCODING_TYPE = 'hex';
+        const vids = generateRandomVIDs(count);
 
-    it('sorted in reversed chronological and alphabetical order', () => {
-        for (let i = 0; i < count; i++) {
-            if (i !== 0) {
-                assert(vids[i - 1] > vids[i],
-                    'previous VersionID is higher than its next');
+        it('sorted in reversed chronological and alphabetical order', () => {
+            for (let i = 0; i < count; i++) {
+                if (i !== 0) {
+                    assert(vids[i - 1] > vids[i],
+                        'previous VersionID is higher than its next');
+                }
             }
-        }
-    });
+        });
 
-    // nodejs 10 no longer returns error for non-hex string versionIds
-    it.skip('should return error decoding non-hex string versionIds', () => {
-        const encoded = vids.map(vid => VID.hexEncode(vid));
-        const decoded = encoded.map(vid => VID.hexDecode(`${vid}foo`));
-        decoded.forEach(result => assert(result instanceof Error));
-    });
+        // nodejs 10 no longer returns error for non-hex string versionIds
+        it.skip('should return error decoding non-hex string versionIds', () => {
+            const encoded = vids.map(vid => VID.hexEncode(vid));
+            const decoded = encoded.map(vid => VID.hexDecode(`${vid}foo`));
+            decoded.forEach(result => assert(result instanceof Error));
+        });
 
-    it('should encode and decode versionIds', () => {
-        const encoded = vids.map(vid => VID.hexEncode(vid));
-        const decoded = encoded.map(vid => VID.hexDecode(vid));
-        assert.strictEqual(vids.length, count);
-        assert.deepStrictEqual(vids, decoded);
-    });
+        it('should encode and decode versionIds', () => {
+            const encoded = vids.map(vid => VID.hexEncode(vid));
+            const decoded = encoded.map(vid => VID.hexDecode(vid));
+            assert.strictEqual(vids.length, count);
```

Comment:
You could get rid of this line as it does not check anything more than the next line and only checks that the original `vids` array indeed has `count` versions in it (so testing the test case in a way)